cpp/src/Slice/Parser.h

```diff
@@ -720,22 +724,25 @@ class Operation : public virtual Contained, public virtual Container
     bool returnsClasses(bool) const;
     bool returnsData() const;
     bool returnsMultipleValues() const;
+    bool hasOutParameters() const;
```

Comment:
I think the non-temporary API should look like:
```
MemberList params() const;
MemberList returnValues() const;
bool hasReturnAndOut() const;
```

returnsClasses and sendsClasses can probabaly be replaced by a simple non-member function:

```
bool containsClass(const MemberList&);
```

returnsData is a synonym for returnValues is not empty? returnsMultipleValues means returnValues size > 1?

Reply:
I've already put refactoring these methods on my eventual TODO list, but for this PR I wanted to keep my changes mostly confined to the Parser and adding return-tuple syntax, and not mess around with the API that we already had in place. I was planning on fixing these when we finish updating the compilers and I was going to remove the old API anyways.

The compilers and parser are perpetually in need of cleaning :) If I fixed everything I found on the spot, my PR's would never be opened!

Reply:
I'll add a method for 'hasReturnAndOut' though like we discussed.